src/pybind/mgr/orchestrator.py

```diff
@@ -380,38 +384,80 @@ def to_json(self):
         return {k:v for (k,v) in out.items() if v is not None}
 
 
+class DeviceSelection(object):
+    def __init__(self, paths=None, id_model=None, size=None, rotates=None, count=None):
+        # type: (List[str], str, int, bool, int) -> ()
+        """
+        ephemeral drive group device specification
+        :param paths: abs paths to the devices.
+        :param id_model: A wildcard string. e.g: "SDD*"
+        :param size: Size specification of format LOW:HIGH.
+            Can also take the the form :HIGH, LOW:
+            or an exact value (as ceph-volume inventory reports)
+        :param rotates: is the drive rotating or not
+        :param count: if this is present limit the number of drives to this number.
+
+        Any attributes (even none) can be included in the device
+        specification structure.
+        """
+        self.paths = paths
+        self.id_model = id_model
+        self.size = size
+        self.rotates = rotates
+        self.count = count
+
+    @classmethod
+    def from_json(cls, device_spec):
+        return cls(**device_spec)
+
+
 class DriveGroupSpec(object):
     """
     Describe a drive group in the same form that ceph-volume
     understands.
     """
-    def __init__(self, devices):
-        self.devices = devices
+    def __init__(self, hosts, data_devices, db_devices=None, wal_devices=None, journal_devices=None,
+                 osds_per_device=None, objectstore='bluestore', encrypted=False, db_slots=None,
+                 wal_slots=None):
+        # type: (Union[List[str], str], DeviceSelection, Optional[DeviceSelection], Optional[DeviceSelection], Optional[DeviceSelection], int, str, bool, int, int) -> ()
 
+        # concept of applying a drive group to a (set) of hosts is tightly
+        # linked to the drive group itself
+        self.hosts = hosts if isinstance(hosts, list) else [hosts]
```

Comment:
I will replace hosts by "target", with the same meaning that we have in the specification ( SUSE specification) . Each orchestrator can obtain easily the list of hosts using the get_inventory method and applying the  "pattern" specified in "target".